pkg/states/state_manager.go

```diff
@@ -0,0 +1,83 @@
+package states
+
+import "github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+
+func Active(userSignup *v1alpha1.UserSignup) bool {
+	return Approved(userSignup) &&
+		!VerificationRequired(userSignup) &&
+		!Deactivated(userSignup)
+}
+
+func Approved(userSignup *v1alpha1.UserSignup) bool {
+	return contains(userSignup.Spec.States, v1alpha1.UserSignupStateApproved)
+}
+
+func SetApproved(userSignup *v1alpha1.UserSignup, val bool) {
+	if val && !contains(userSignup.Spec.States, v1alpha1.UserSignupStateApproved) {
```

Comment:
Shouldn't we replace some states by new ones instead of always appending the new one?

For example, when setting to Approved we should remove VerificationRequired. When setting Deactivated we should remove Approved.

Reply:
Sure thing, we just need to define the business rules here.  I've implemented the approved and deactivated cases already as you describe.